astroquery/vo/core.py

```diff
@@ -0,0 +1,237 @@
+        ----------
+        service_type : string, required
+            Valid VO service types are: "conesearch", "simpleimageaccess", "simplespectralaccess", "tableaccess"
+            They may be shortened to "cone", "image", "spectr", or "table" or "tap", respectively.
+        keyword : string, optional
+            Default is None.
+            The query will return any services which contain this keyword in their ivoid, title, or description.
+        waveband : string, optional
+            Comma-delimited list of wavebands.  Results will contain all services that offer at least one of these wavebands.
+            Legal values are listed below.
+        source : string, optional
+            Any substring in ivoid (a unique identifier of the service)
+        publisher : string, optional
+            The name of any publishing organization (e.g., "stsci", "heasarc")
+        order_by : string, optional
+            What column to order it by.  The returned columns are:
+                "waveband","short_name","ivoid","res_description","access_url","reference_url","publisher", service_type"
+        logic_string : string, optional
+            Any other string you want to add to the ADQL where clause, should start with " and ".
+        verbose : bool, optional
+            Default is False.
+            When True, the ADQL query computed from the keyword arguments will be printed.
+        """
+
+        adql = self._build_adql(**kwargs)
+        if adql is None:
+            raise ValueError('Unable to compute query based on input arguments.')
+
+        if kwargs.get('verbose'):
+            print('Registry:  sending query ADQL = {}\n'.format(adql))
+
+        url = self._REGISTRY_TAP_SYNC_URL
+
+        tap_params = {
+            "request": "doQuery",
+            "lang": "ADQL",
+            "query": adql
+        }
+
+        response = utils.try_query(url, post_data=tap_params, timeout=self._TIMEOUT, retries=self._RETRIES)
+
+        if kwargs.get('verbose'):
+            print('Queried: {}\n'.format(response.url))
+
+        aptable = utils.astropy_table_from_votable_response(response)
+        return aptable
+
+    def _build_adql(self, **kwargs):
+
+        # Default values
+        service_type = ""
+        keyword = ""
+        waveband = ""
+        source = ""
+        publisher = ""
+        order_by = ""
+        logic_string = " and "
+
+        # Find the keywords we recognize
+        for key, val in kwargs.items():
+            if key == 'service_type':
+                service_type = val
+            elif key == 'keyword':
+                keyword = val
+            elif key == 'waveband':
+                waveband = val
+            elif key == 'source':
+                source = val
+            elif key == 'publisher':
+                publisher = val
+            elif key == 'order_by':
+                order_by = val
+            elif key == 'logic_string':
+                logic_string = val
```

Comment:
These can all be specified much more easily as actual keyword arguments

Reply:
Addressed: kwargs was removed in favor of explicit keywords.